src/foremast/securitygroup/create_securitygroup.py

```diff
@@ -194,11 +194,21 @@ def add_cidr_rules(self, rules):
 
         return True
 
+    def resolve_self_references(self, rules):
+        """Resolves `$self` references to actual application name in security group rules."""
+        resolved_rules = {}
+        for app, rule in rules.items():
```

Comment:
I think this would be easier with a `rule = dict.get(); if rule:`.

Reply:
Are you suggesting something like this:
```
for app in rules:
  if app == '$self':
    app = self.app_name
  resolved_rules[app] = rules.get(app)
```
or
```
rule = rules.pop('$self')
if rule:
   rules[self.app_name] = rule
return rules
```

Reply:
Second one. Might as well use the `dict` features.